clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java

```diff
@@ -923,11 +923,12 @@ else if (def.type == Type.STRING && def.defaultValue.toString().isEmpty())
      * documentation.
      */
     public String toRst() {
-        List<ConfigKey> configs = sortedConfigs();
+        List<ConfigKey> configs = sortedConfigsInGroup();
         StringBuilder b = new StringBuilder();
 
         for (ConfigKey def : configs) {
             b.append("``");
+            b.append(def.group);
```

Comment:
this doesn't seem right -- the JIRA suggests that the group should be a section header for all config keys with that group